### PR TITLE
REL-2791: ephemeris plot bugfixes

### PR DESCRIPTION
**Pull request by swalker2m**
A couple of tweaks to ephemeris plotting to remove erroneous points:

1) An update to ancient world coordinate to screen coordinate conversion code to remove points from the opposite side of the sky.

2) Filtering out points that don't fall on the viewport of the image widget.

### CODE REVIEW

File: bundle/jsky.app.ot/src/main/scala/jsky/app/ot/tpe/feat/TpeEphemerisFeature.scala [hunk 1]
```diff
@@ -85,17 +95,19 @@ class TpeEphemerisFeature extends TpeImageFeature("Ephemeris", "Show interpolate
...
     obsScreenEphemeris(e).foreach { se =>
       g.setColor(Color.GREEN)
-      drawTrack(g, se)
-      se.takeRight(2) match {
-        case List((_, p1), (t, p2)) => drawLabel(g, t, p2, ang(p1, p2))
-        case _ => // degenerate ephemeris; no track was drawn, ignore
+      se.foreach(drawTrack(g, _))
+      se.foreach { lst =>
+        lst.takeRight(2) match {
+          case List((_, p1), (t, p2)) => drawLabel(g, t, p2, ang(p1, p2))
```
Comment (cquiroz) [inline bundle/jsky.app.ot/src/main/scala/jsky/app/ot/tpe/feat/TpeEphemerisFeature.scala]: `ScreenEphemeris` is a alias of a tuple? Screen coordinates?

File: bundle/jsky.coords/src/main/java/jsky/coords/platepos.java [hunk 2]
```diff
@@ -56,7 +56,7 @@
 public class platepos {
 
     // Terms used by all methods
-    private static final double COND2R = 1.745329252e-2;
+    private static final double COND2R = Math.PI / 180.0;
```
Comment (cquiroz) [inline bundle/jsky.coords/src/main/java/jsky/coords/platepos.java]: 👍

**Comment by cquiroz** [comment]
Looks good to me :+1:

File: bundle/jsky.app.ot/src/main/scala/jsky/app/ot/tpe/feat/TpeEphemerisFeature.scala (hunk 1 above)
Comment (swalker2m) [inline bundle/jsky.app.ot/src/main/scala/jsky/app/ot/tpe/feat/TpeEphemerisFeature.scala]:
````scala
type ScreenEphemeris = List[(Long, Point)]
````

So it is a `List` of times and the corresponding location on the screen.  `onScreenEphemeris` was modified to return a `List[ScreenEphemeris]` which is then a `List[List[(Long, Point)]]`. Each `ScreenEphemeris` entry in the list represents a continuous section of ephemeris elements for which we have a valid screen location.

**Comment by tpolecat** [comment]
This looks good. Nice sleuthing!

**Pull request merged**